specs/gloas/beacon-chain.md

```diff
@@ -362,15 +362,16 @@ def has_compounding_withdrawal_credential(validator: Validator) -> bool:
 ```python
 def is_attestation_same_slot(state: BeaconState, data: AttestationData) -> bool:
     """
-    Checks if the attestation was for the block proposed at the attestation slot.
+    Check if the attestation is for the block proposed at the attestation slot.
     """
     if data.slot == 0:
         return True
-    is_matching_blockroot = data.beacon_block_root == get_block_root_at_slot(state, Slot(data.slot))
-    is_current_blockroot = data.beacon_block_root != get_block_root_at_slot(
-        state, Slot(data.slot - 1)
-    )
-    return is_matching_blockroot and is_current_blockroot
+
+    blockroot = data.beacon_block_root
+    curr_blockroot = get_block_root_at_slot(state, Slot(data.slot))
```

Comment:
This is more than a nit: I've seen people confused and errors in tests because of thinking that the root at `data.slot` is the current blockroot, making the attestation for the current block as if it were for `state.slot` and this causes problems. The block is the block that was the current block at the attestation deadline. So perhaps instead of `curr_blockroot` we can use `timely_blockroot`? or something that isn't "current"?